src/host/srvinit.cpp

```diff
@@ -871,7 +890,11 @@ DWORD WINAPI ConsoleIoThread(LPVOID lpParameter)
     // If we were given a message on startup, process that in our context and then continue with the IO loop normally.
     if (lpParameter)
     {
-        ReceiveMsg = *(PCONSOLE_API_MSG)lpParameter;
+        // Capture the incoming lpParameter into a unique_ptr so we can appropriately
+        // free the heap memory when we're done getting the important bits out of it below.
+        std::unique_ptr<CONSOLE_API_MSG> capturedMessage{ static_cast<PCONSOLE_API_MSG>(lpParameter) };
+
+        ReceiveMsg = *capturedMessage.get();
         ReceiveMsg._pApiRoutines = &globals.api;
         ReceiveMsg._pDeviceComm = globals.pDeviceComm;
         IoSorter::ServiceIoOperation(&ReceiveMsg, &ReplyMsg);

```

Comment:
Meh, can we just patch `capturedMessage` directly and hand it off as a pointer instead of copying it?

```suggestion
        IoSorter::ServiceIoOperation(capturedMessage.get(), &ReplyMsg);
```

Reply:
Having the leftover stack state in `ReceiveMsg` and `ReplyMsg` when the dump I was given was hung later on at `HRESULT hr = ServiceLocator::LocateGlobals().pDeviceComm->ReadIo(ReplyMsg, &ReceiveMsg);` was what lead me to understand this bug to begin with.

As such, I tried to maintain keeping the "most recent message before hang" in the local variables. If I use the heap one directly, it'll disappear to space if I get another similar hang dump and I'll only have the reply to go on. Additionally, the api and devicecomm bits need to be patched up in `ReceiveMsg` anyway, no matter what, as they're reused message over message.